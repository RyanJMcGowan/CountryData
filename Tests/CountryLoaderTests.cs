using CountryData;
using ObjectApproval;
using Xunit;

public class CountryLoaderTests
{
    [Fact]
    public void LoadLocationData()
    {
        var states = CountryLoader.LoadLocationData("PW");
        ObjectApprover.VerifyWithJson(states);
    }

    [Fact]
    public void LoadSpecificLocationData()
    {
        var states = CountryLoader.LoadPalauLocationData();
        ObjectApprover.VerifyWithJson(states);
    }

    [Fact]
    public void LoadAll()
    {
        CountryLoader.LoadAll();
        Assert.NotEmpty(CountryLoader.LoadedLocationData);
    }
}