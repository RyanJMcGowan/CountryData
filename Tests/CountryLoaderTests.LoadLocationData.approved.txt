{
  "$type": "Country",
  "Name": "Palau",
  "Iso": "PW",
  "Iso3": "PLW",
  "IsoNumeric": 585,
  "Fips": "PS",
  "Capital": "Melekeok",
  "Area": 458.0,
  "Continent": "OC",
  "TopLevelDomain": ".pw",
  "CurrencyCode": "USD",
  "CurrencyName": "Dollar",
  "PhonePrefix": "680",
  "PostCodeFormat": "96940",
  "PostCodeRegex": "^(96940)$",
  "Languages": {
    "$type": "List<string>",
    "$values": [
      "pau",
      "sov",
      "en-PW",
      "tox",
      "ja",
      "fil",
      "zh"
    ]
  },
  "States": {
    "$type": "List<State>",
    "$values": [
      {
        "$type": "State",
        "Provinces": {
          "$type": "List<Province>",
          "$values": [
            {
              "$type": "Province",
              "Communities": {
                "$type": "List<Community>",
                "$values": [
                  {
                    "$type": "Community",
                    "Places": {
                      "$type": "List<Place>",
                      "$values": [
                        {
                          "$type": "Place",
                          "Name": "Palau",
                          "PostCode": "96940",
                          "LatLong": "7.2257,134.3622"
                        }
                      ]
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    ]
  }
}